external_apps/bid/bid_action/bidder_decision.go

```diff
@@ -0,0 +1,204 @@
+	}
+
+	return true, nil
+}
+
+func (b BidderDecisionTx) ProcessCheck(ctx *action.Context, tx action.RawTx) (bool, action.Response) {
+	ctx.Logger.Detail("Processing BidderDecision Transaction for CheckTx", tx)
+	return runBidderDecision(ctx, tx)
+}
+
+func (b BidderDecisionTx) ProcessDeliver(ctx *action.Context, tx action.RawTx) (bool, action.Response) {
+	ctx.Logger.Detail("Processing BidderDecision Transaction for DeliverTx", tx)
+	return runBidderDecision(ctx, tx)
+}
+
+func (b BidderDecisionTx) ProcessFee(ctx *action.Context, signedTx action.SignedTx, start action.Gas, size action.Gas) (bool, action.Response) {
+	return action.BasicFeeHandling(ctx, signedTx, start, size, 1)
+}
+
+func runBidderDecision(ctx *action.Context, tx action.RawTx) (bool, action.Response) {
+	bidderDecision := BidderDecision{}
+	err := bidderDecision.Unmarshal(tx.Data)
+	if err != nil {
+		return helpers.LogAndReturnFalse(ctx.Logger, action.ErrWrongTxType, bidderDecision.Tags(), err)
+	}
+
+	//1. verify bidConvId exists in ACTIVE store
+	bidMasterStore, err := GetBidMasterStore(ctx)
+	if err != nil {
+		return helpers.LogAndReturnFalse(ctx.Logger, bid_data.ErrGettingBidMasterStore, bidderDecision.Tags(), err)
+	}
+	if !bidMasterStore.BidConv.WithPrefixType(bid_data.BidStateActive).Exists(bidderDecision.BidConvId) {
+		return helpers.LogAndReturnFalse(ctx.Logger, bid_data.ErrBidConvNotFound, bidderDecision.Tags(), err)
+	}
+	bidConv, err := bidMasterStore.BidConv.WithPrefixType(bid_data.BidStateActive).Get(bidderDecision.BidConvId)
+	if err != nil {
+		return helpers.LogAndReturnFalse(ctx.Logger, bid_data.ErrGettingBidConv, bidderDecision.Tags(), err)
+	}
+
+	//2. check expiry
+	deadLine := time.Unix(bidConv.DeadlineUTC, 0)
+
+	if deadLine.Before(ctx.Header.Time.UTC()) {
+		return helpers.LogAndReturnFalse(ctx.Logger, bid_data.ErrExpiredBid, bidderDecision.Tags(), err)
+	}
+
+	//3. check asset availability
+	available, err := IsAssetAvailable(ctx, bidConv.AssetName, bidConv.AssetType, bidConv.AssetOwner)
+	if err != nil || available == false {
+		return helpers.LogAndReturnFalse(ctx.Logger, bid_data.ErrInvalidAsset, bidderDecision.Tags(), err)
+	}
+
+	//4. check bidder's identity
+	if !bidderDecision.Bidder.Equal(bidConv.Bidder) {
+		return helpers.LogAndReturnFalse(ctx.Logger, bid_data.ErrWrongBidder, bidderDecision.Tags(), err)
+	}
+
+	//5. get the active counter offer
+	activeOffer, err := bidMasterStore.BidOffer.GetActiveOffer(bidderDecision.BidConvId, bid_data.TypeCounterOffer)
+	if err != nil || activeOffer == nil {
+		return helpers.LogAndReturnFalse(ctx.Logger, bid_data.ErrGettingActiveCounterOffer, bidderDecision.Tags(), err)
+	}
+
+	//6. if reject
+	if bidderDecision.Decision != bid_data.RejectBid && bidderDecision.Decision != bid_data.AcceptBid {
+		return helpers.LogAndReturnFalse(ctx.Logger, bid_data.ErrInvalidBidderDecision, bidderDecision.Tags(), err)
+	} else if bidderDecision.Decision == bid_data.RejectBid {
```

Comment:
any particular reason we need the else if? 
an if would be more idiomatic

Reply:
yeah sure